src/app/core/service/alert.service.ts

```diff
@@ -112,6 +112,14 @@ export class AlertService {
     return this.alert(AlertLocation.MAIN, AlertType.DANGER, `(${payload.response.status}) ${payload.response.message}`, true, 15000);
   }
 
+  public getNetworkFailureAlert(payload: { response: any }): fromAlert.AlertActions {
+    return this.alert(AlertLocation.MAIN, AlertType.DANGER, `(${payload.response.status}) ${payload.response.message}`, true, 15000);
+  }
+
+  public getCoInvestigatorNetworkFailureAlert(payload: { response: any }): fromAlert.AlertActions {
+    return this.alert(AlertLocation.MAIN, AlertType.DANGER, `(${payload.response.status}) ${payload.response.message}`, true, 15000);
+  }
+
   public findByIdInFailureAlert(payload: { response: any }): fromAlert.AlertActions {
     return this.alert(AlertLocation.MAIN, AlertType.DANGER, `(${payload.response.status}) ${payload.response.message}`, true, 15000);
   }
```

Comment:
A lot of these are the same, except for the method names.

Is it possible to use the same function rather than have multiple identical functions with different names?
It just seems like a lot of redundancy.

I see this is the current pattern with the program.

Reply:
Yeah, I have been thinking about these and think there is an issue somewhere to improve front end error handling. I am going to defer this cleanup for now.